debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -388,6 +380,46 @@ public Lsn getFromLsn() {
                                 columns.get(changeTableObjectId)));
             }
             return changeTables;
+        };
+
+        String query = GET_LIST_OF_CDC_ENABLED_TABLES.replace(DATABASE_NAME_PLACEHOLDER, databaseName);
+        if ((fromLsn.getBinary() == null && toLsn.getBinary() == null)) {
+            return queryAndMap(query, mapper);
+        }
+        else if (fromLsn.getBinary() == null) {
+            query = query + " WHERE ct.start_lsn <= ?";
+            return prepareQueryAndMap(query, ps -> {
+                ps.setBytes(1, toLsn.getBinary());
+            }, mapper);
+        }
+        else if (toLsn.getBinary() == null) {
+            query = query + " WHERE ct.end_lsn >= ? OR ct.end_lsn IS NULL";
+            return prepareQueryAndMap(query, ps -> {
+                ps.setBytes(1, fromLsn.getBinary());
+            }, mapper);
+        }
+        else {
+            query = query + " WHERE ct.start_lsn <= ? AND (ct.end_lsn >= ? OR ct.end_lsn IS NULL)";
```

Comment:
I think it should be `>=` because if context contains left LSN we select the next batch using `TOP + 1`

Reply:
or maybe I misinterpret that logic, tests pass anyway
at the same time in the other place we use `BETWEEN` which is inclusive

Reply:
From https://github.com/sugarcrm/debezium/pull/50:

> `max.iteration.transactions = 1` won't work because of the combination of `>=` and `TOP 1`.

I don't remember what exactly it means 😬 

IIRC, this test would fail without the `+1`:
https://github.com/sugarcrm/debezium/blob/279ae15e26ff021ee0646b638daf016526af1635/debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java#L2558-L2568

Originally, it was implemented as: https://github.com/sugarcrm/debezium/blob/f96e210638f68adc18522ff18c062572defa756b/debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java#L2429